packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -130,11 +137,37 @@ export default createRule<[Options], MessageIds>({
 
           const option = options[fullDirective];
           if (option === true) {
-            context.report({
-              data: { directive },
-              node: comment,
-              messageId: 'tsDirectiveComment',
-            });
+            if (directive === 'ignore' && options['ts-expect-error'] !== true) {
+              // Special case to suggest @ts-expect-error instead of @ts-ignore,
+              // as long as @ts-expect-error is banned outright.
```

Comment:
I would consider `// @ts-expect-error` to still be superior over `// @ts-ignore` even if it's banned too. Sometimes I have to use it with an eslint-disable comment, but it's still preferable. I'd say let's always suggest it.